test/OpenTelemetry.Api.Tests/Trace/TracerTest.cs

```diff
@@ -404,7 +404,10 @@ static void InnerTest()
                 getTracerThread.Join();
             }
 
-            Assert.Empty(tracers);
+            foreach (var kvp in tracers)
```

Comment:
@pentp Can you explain why this test is changing? For simple cleanup/simplification work I would expect all the tests to pass as they did before so this worries me a bit 😄

Reply:
It's changing because I removed the `ConcurrentDictionary.Clear()` call [here](https://github.com/open-telemetry/opentelemetry-dotnet/pull/6048/files#diff-ae497458655ac369916b974061ceb0f1245350594178410cefa3d4f045cef83b).

It's unnecessary from a functional perspective (the dictionary isn't used in any way after the class field is nulled out), but somewhat expensive - it takes all the locks and allocates new backing arrays which can be potentially quite large: https://source.dot.net/#System.Collections.Concurrent/System/Collections/Concurrent/ConcurrentDictionary.cs,108c257ef528517e

Reply:
Should I restore the `Clear()` call?

Reply:
Yes, it's better to restore the existing functionality as this is a cleanup / simplification PR.